tendermint/src/time.rs

```diff
@@ -16,33 +18,35 @@ use crate::error::Error;
 /// Tendermint timestamps
 /// <https://github.com/tendermint/spec/blob/d46cd7f573a2c6a2399fcab2cde981330aa63f37/spec/core/data_structures.md#time>
 #[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
-#[serde(try_from = "Timestamp", into = "Timestamp")]
-pub struct Time(DateTime<Utc>);
+#[serde(try_from = "Timestamp")]
+pub struct Time(pub DateTime<Utc>);
 
 impl Protobuf<Timestamp> for Time {}
 
-impl From<Timestamp> for Time {
-    fn from(value: Timestamp) -> Self {
-        // prost_types::Timestamp has a SystemTime converter but
-        // tendermint_proto::Timestamp can be JSON-encoded
-        let prost_value = prost_types::Timestamp {
-            seconds: value.seconds,
-            nanos: value.nanos,
-        };
+impl TryFrom<Timestamp> for Time {
+    type Error = Error;
 
-        SystemTime::from(prost_value).into()
+    fn try_from(value: Timestamp) -> Result<Self, Error> {
```

Comment:
I implement `TryFrom` instead of `From` to allow safe conversion of the raw protobuf `Timestamp` to the `Time` type. This is necessary as one is using `i32` while the other use `u32`. Although the overflow is unlikely to occur, for security purpose it is still possible that an attacker may try to provide overflow time values to cause undefined behavior on the relayer or the chain. So it is better to return an error in case that happens.

As a result of this change, a number of `From` instances for various data structures that use `Time` have to be updated to implement `TryFrom` instead.